extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -423,36 +423,57 @@ static public function setmainDomain ( $city_id, $domain = null, $reason = null
 	 * @access public
 	 * @static
 	 *
-	 * @param $url string - domain name
+	 * @param $url string - domain name in form:
+	 * - http://www.community-name.wikia.com
+	 * - http://community-name.wikia.com
+	 * - www.community-name.wikia.com
+	 * - community-name.wikia.com
+	 * - community-name
 	 *
 	 * @return integer - id of domain or null if not found
 	 */
 	static public function UrlToID( $url ) {
-
 		$city_id = false;
+
+		$url = self::prepareUrlToParse( $url );
 		$parts = parse_url( $url );
+
 		if ( isset( $parts[ "host" ] ) ) {
 			$host = self::getDomainHash( $parts[ "host" ] );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)